beacon-chain/blockchain/service.go

```diff
@@ -186,17 +187,12 @@ func (c *ChainService) StateInitializedFeed() *event.Feed {
 
 // ChainHeadRoot returns the hash root of the last beacon block processed by the
 // block chain service.
-func (c *ChainService) ChainHeadRoot() ([32]byte, error) {
-	head, err := c.beaconDB.ChainHead()
+func (c *ChainService) ChainHeadRoot(state *pb.BeaconState) ([32]byte, error) {
+	root, err := b.BlockRoot(state, state.Slot-1)
```

Comment:
we should be careful of this, imagine the following the scenario:
**validator current time:** 25s
**beacon node state slot:** 3
Beacon node has just processed the head for slot 3, validator wants whatever the head is at slot 3 but given `state.Slot-1`, validator can only get the block root at slot 2 but it should be 3

Reply:
Hey @nisdas is this still relevant?

Reply:
Bump @nisdas